test/e2e/specs/realrp/vnetpeering.go

```diff
@@ -0,0 +1,129 @@
+package realrp
+
+import (
+	"context"
+	"errors"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-06-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp/client"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+)
+
+var _ = Describe("Peer Vnet tests [Vnet]", func() {
+	var (
+		ctx          = context.Background()
+		cli          *azure.Client
+		rg           = os.Getenv("RESOURCEGROUP")
+		vnetPeerName = "vnetPeer"
+		region       = os.Getenv("AZURE_REGION")
+		clientID     = os.Getenv("AZURE_CLIENT_ID")
+		clientSecret = os.Getenv("AZURE_CLIENT_SECRET")
+	)
+
+	BeforeEach(func() {
+		var err error
+		cli, err = azure.NewClientFromEnvironment(false)
+		Expect(err).ToNot(HaveOccurred())
+		if os.Getenv("AZURE_REGION") == "" {
+			Expect(errors.New("AZURE_REGION is not set")).ToNot(HaveOccurred())
+		}
+		if os.Getenv("RESOURCEGROUP") == "" {
+			Expect(errors.New("RESOURCEGROUP is not set")).ToNot(HaveOccurred())
+		}
+	})
+
+	AfterEach(func() {
+		result, _ := cli.Groups.Delete(ctx, rg)
```

Comment:
You shouldn't need to delete. Today, we are cleaning everything up in the `teardown` container of the e2e template. We may want to move resource deletion in the e2e framework in the future but we should be doing that holistically.

Reply:
Hrm, actually you may need to keep this for now. I will have another pass tomorrow and think about this more.

Reply:
Yeah, you can probably keep this, it would make the test more self-contained. Eventually, we should do this for all other e2e tests.